### DISCUSSION TITLE
Global Variables

### DISCUSSION THREAD

**Discussion by simsrw73** [JanDeDobbeleer/oh-my-posh#3266] [Ideas]
Is there any way to have global variables that can be accessed from any segment template? I want to be able to customize my theme without having to duplicate it. Possible usage scenarios:

1. Define chars to be referenced and printed in various templates.
2. Define a value that will be tested in a template conditional.
3. Maybe, like palettes, define multiple sets of vars with a selector that makes one set active.
4. Maybe, a way to select or specify values from the omp init.

I tried a number of things but couldn't get anything to work. Is there something like this already baked in? Or can some feature be (mis)used to accomplish the same goal?

E.g. I want to allow user to select whether to have angle (>) or slant (/) separators between segments. And to select angle (>), slant (/), or rounded (◗) end caps, where there are right and left-side variations.

**Comment by JanDeDobbeleer** [comment]
@simsrw73 no, there's no such thing today. Also very exotic as a use-case. I can give it a thought.

**Reply by JanDeDobbeleer** [reply]
@simsrw73 https://github.com/JanDeDobbeleer/oh-my-posh/pull/3517, need to have a good look tomorrow at the implementation as something did feel off.

**Comment by simsrw73** [comment]
Sorry. I haven't got a chance to look at this yet. I actually haven't done anything on my omp themes since posting... I upgraded my PC, had trouble with it crashing, had to reinstall Windows, and everything is still in pieces. But I am coming back to this soon. It looks perfect. Thank you!!!
